react@"^17.0.1" from the root project, peer react@"^0.14.9 || ^15.3.0 || ^16.0.0" from nuka-carousel@4.7.7

# Bugs and Questions

### Prerequisites

> [max@workstation m]$ npm install

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: shahro@0.1.0
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^0.14.9 || ^15.3.0 || ^16.0.0" from nuka-carousel@4.7.7
npm ERR! node_modules/nuka-carousel
npm ERR!   nuka-carousel@"^4.7.5" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/maxb/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/maxb/.npm/_logs/2021-04-30T13_32_25_424Z-debug.log
```

> [max@workstation m]$ cat package.json 
```
{
  "name": "shahro",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@craco/craco": "^6.1.1",
    "@material-ui/core": "^4.11.3",
    "@material-ui/icons": "^4.11.2",
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "axios": "^0.21.1",
    "nuka-carousel": "^4.7.5",
    "react": "^17.0.1",
    "react-content-loader": "^6.0.2",
    "react-dom": "^17.0.1",
    "react-helmet": "^6.1.0",
    "react-indiana-drag-scroll": "^2.0.1",
    "react-modern-calendar-datepicker": "^3.1.6",
    "react-redux": "^7.2.2",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "react-spring": "^8.0.27",
    "react-swipeable-bottom-sheet": "^1.1.2",
    "react-use-gesture": "^9.1.3",
    "react-verification-code-input": "^1.2.9",
    "redux": "^4.0.5",
    "redux-thunk": "^2.3.0",
    "styled-components": "^5.2.1",
    "web-vitals": "^1.0.1"
  },
  "scripts": {
    "start": "craco start",
    "build": "craco build",
    "test": "craco test",
    "eject": "craco eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.13.10",
    "@babel/preset-env": "^7.13.10",
    "@tailwindcss/postcss7-compat": "^2.0.3",
    "autoprefixer": "^9.8.6",
    "postcss": "^7.0.35",
    "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.0.3"
  }
}
```

--- a/package.json
+++ b/package.json
@@ -42,7 +42,7 @@
     "prettier": "2.1.2",
     "puppeteer": "^5.5.0",
     "react": "^16.9.0",
-    "react-dom": "^16.0.0",
+    "react-dom": "^16.9.0",
     "tslint": "^5.12.0",
     "typescript": "^3.6.2",
     "url-loader": "^4.1.1",
@@ -51,8 +51,8 @@
     "webpack-dev-server": "^3.11.0"
   },
   "peerDependencies": {
-    "react": "^0.14.9 || ^15.3.0 || ^16.0.0",
-    "react-dom": "^0.14.9 || ^15.3.0 || ^16.0.0"
+    "react": "^17.0.1",
+    "react-dom": "^17.0.1"
   },
   "resolutions": {
     "jsdom": "^14.0.0"

--- a/package.json
+++ b/package.json
@@ -51,8 +51,8 @@
     "webpack-dev-server": "^3.11.0"
   },
   "peerDependencies": {
-    "react": "^17.0.1",
-    "react-dom": "^17.0.1"
+    "react": "^0.14.9 || ^15.3.0 || ^16.0.0 || ^17.0.1",
+    "react-dom": "^0.14.9 || ^15.3.0 || ^16.0.0 || ^17.0.1"
   },
   "resolutions": {
     "jsdom": "^14.0.0"

